src/parser/grammar.y

```diff
@@ -36,33 +36,50 @@
 query ::= expr(A). { ctx->root = A; }
 
 expr(A) ::= matchClause(B) whereClause(C) createClause(D) returnClause(E) orderClause(F) limitClause(G). {
-	A = New_AST_Query(B, C, D, NULL, NULL, NULL, E, F, G);
+	A = New_AST_Query(B, C, D, NULL, NULL, NULL, E, F, G, NULL);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) createClause(D). {
-	A = New_AST_Query(B, C, D, NULL, NULL, NULL, NULL, NULL, NULL);
+	A = New_AST_Query(B, C, D, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) deleteClause(D). {
-	A = New_AST_Query(B, C, NULL, NULL, NULL, D, NULL, NULL, NULL);
+	A = New_AST_Query(B, C, NULL, NULL, NULL, D, NULL, NULL, NULL, NULL);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) setClause(D). {
-	A = New_AST_Query(B, C, NULL, NULL, D, NULL, NULL, NULL, NULL);
+	A = New_AST_Query(B, C, NULL, NULL, D, NULL, NULL, NULL, NULL, NULL);
 }
 
 expr(A) ::= matchClause(B) whereClause(C) setClause(D) returnClause(E) orderClause(F) limitClause(G). {
-	A = New_AST_Query(B, C, NULL, NULL, D, NULL, E, F, G);
+	A = New_AST_Query(B, C, NULL, NULL, D, NULL, E, F, G, NULL);
 }
 
 expr(A) ::= createClause(B). {
-	A = New_AST_Query(NULL, NULL, B, NULL, NULL, NULL, NULL, NULL, NULL);
+	A = New_AST_Query(NULL, NULL, B, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
 }
 
 //expr(A) ::= mergeClause(B). {
-//	A = New_AST_Query(NULL, NULL, NULL, B, NULL, NULL, NULL, NULL, NULL);
+//	A = New_AST_Query(NULL, NULL, NULL, B, NULL, NULL, NULL, NULL, NULL, NULL);
 //}
 
+expr(A) ::= indexOpToken(B) INDEX ON indexLabel(C) indexProp(D) . {
+	A = New_AST_Query(NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, AST_IndexOp(C.strval, D.strval, B));
+}
+
+%type indexOpToken { AST_IndexOpType }
+
+indexOpToken(A) ::= CREATE . { A = CREATE_INDEX; }
+indexOpToken(A) ::= DROP . { A = DROP_INDEX; }
```

Comment:
I agree with your idea of simplifying the index rules - does this seem like an appropriate solution?

I couldn't think of how to handle CREATE and DROP in a single rule, so instead took this approach (following the style of the `relation(A)` rules used for Filters).

If you have a simpler solution, I'd be happy to implement it!

Reply:
This is perfect.